msm/src/circuit_design/capabilities.rs

```diff
@@ -45,3 +45,59 @@ where
 {
     fn lookup(&mut self, lookup_id: LT, value: &Self::Variable);
 }
+
+////////////////////////////////////////////////////////////////////////////
+// Helpers
+////////////////////////////////////////////////////////////////////////////
+
+/// Write an array of values simultaneously.
+pub fn read_column_array<F, Env, const ARR_N: usize, CIx: ColumnIndexer, Func>(
+    env: &mut Env,
+    column_foo: Func,
```

Comment:
I suppose `column_mapping`?

Reply:
Renamed into `column_map` here and elsewhere.